src/routes/formsgSiteLaunch.ts

```diff
@@ -0,0 +1,212 @@
+    )
+
+    // 2. Check arguments
+    if (!requesterEmail) {
+      // Most errors are handled by sending an email to the requester, so we can't recover from this.
+      throw new BadRequestError(
+        "Required 'Government E-mail' input was not found"
+      )
+    }
+
+    if (!agencyEmail) {
+      // Most errors are handled by sending an email to the requester, so we can't recover from this.
+      throw new BadRequestError("Required 'Agency E-mail' input was not found")
+    }
+
+    try {
+      if (!primaryDomain) {
+        const err = `A primary domain is required`
+        await this.sendLaunchError(
+          [requesterEmail, agencyEmail],
+          repoName,
+          submissionId,
+          err
+        )
+        return res.sendStatus(200)
+      }
+      if (!repoName) {
+        const err = `A repository name is required`
+        await this.sendLaunchError(
+          [requesterEmail, agencyEmail],
+          repoName,
+          submissionId,
+          err
+        )
+        return res.sendStatus(200)
+      }
+
+      const agencyUser = await this.usersService.findByEmail(agencyEmail)
+      const requesterUser = await this.usersService.findByEmail(requesterEmail)
+
+      if (!agencyUser) {
+        const err = `Form submitter ${agencyEmail} is not an Isomer user. Register an account for this user and try again.`
+        await this.sendLaunchError(
+          [requesterEmail, agencyEmail],
+          repoName,
+          submissionId,
+          err
+        )
+        return res.sendStatus(200)
+      }
+
+      if (!requesterUser) {
+        const err = `Form submitter ${requesterUser} is not an Isomer user. Register an account for this user and try again.`
+        await this.sendLaunchError(
+          [requesterEmail, agencyEmail],
+          repoName,
+          submissionId,
+          err
+        )
+        return res.sendStatus(200)
+      }
+
+      // 3. Use service to Launch site
+      // note: this function is not be async due to the timeout for http requests.
+      const launchSite = this.infraService.launchSite(
+        submissionId,
+        requesterUser,
+        agencyUser,
+        repoName,
+        primaryDomain,
+        subDomainSettings
+      )
+
+      // only send success message after promise has been resolved
+      launchSite.then(async () => {
+        await this.sendLaunchSuccess(
+          // fields below are guarenteed to be a string due to prior checks
+          <string>requesterEmail,
+          <string>repoName,
+          submissionId
+        )
+      })
+    } catch (err) {
+      await this.sendLaunchError(
+        [requesterEmail, agencyEmail],
+        repoName,
+        submissionId,
+        `Error: ${err}`
+      )
+      logger.error(err)
+    }
+
+    return res.sendStatus(200)
+  }
+
+  sendLaunchError = async (
+    email: string[],
```

Comment:
This is fair - I have made the changes here: eaa5daa
may I clarify what you mean by `responses` here? Are you referring to `errorMessages` instead?

Reply:
oh i meant the form's `responses`! it doesn't matter if the intended consumer is us - because we own the form, we only have to lookup the submission id (albeit inconvenient). but for emails that we also send to the agency user, they won't be able to obtain useful information beyond the error message ._.

Reply:
I have removed emailing to agencies [completely](https://github.com/isomerpages/isomercms-backend/pull/558/commits/ee680c5050ae7adffd33301291fa1ec8af450b77) . The recent site launches taught us that there are a lot of external processes and things that might go wrong, and this might warrant manual human interaction. 
For now only ops to receive the emails, still in discussion with them on how to refine this process, we can keep it as this for now and change this in the future.